### PR TITLE
392 Unexpected Gain using high shelve filter with shelve type I and II

### LINKED ISSUE #392: Unexpected Gain using high shelve filter with shelve type I and II
## General

* pyfar version: 0.5.1
* Python version: 3.10.4
* Operating System: MacOS 12.5.1
* Did you install pyfar via pip: yes

## Description

Using `pf.dsp.filter.high_shelve()` with the following explanation for the shelve type
```
shelve_type : str
Defines the characteristic frequency. The default is ``'I'``
    ``'I'``
        defines the characteristic frequency 3 dB below the gain value if
        the gain is positive and 3 dB above the gain value otherwise
    ``'II'``
        defines the characteristic frequency at 3 dB if the gain is
        positive and at -3 dB if the gain is negative.
    ``'III'``
        defines the characteristic frequency at gain/2 dB.
```
Using the high shelve filter with a gain of -6 dB and a characteristic frequency of 1 kHz I expected the gain at 1 kHz to be -3 dB, regardless of the shelve type. But the gain at 1 kHz turns out to be -4 dB using Type I and -2 dB using Type II.

## What I Did

```
import pyfar as pf
fs = 44100
frequency = 1000
gain = -6
order = 2

impulse = pf.signals.impulse(n_samples=fs, sampling_rate=fs)
for shelve_type in ['I', 'II', 'III']:
    ir = pf.dsp.filter.high_shelve(signal=impulse, frequency=frequency, gain=gain, order=order, 
                                   shelve_type=shelve_type)
    pf.plot.freq(ir, label=f'shelve type {shelve_type}')

plt.ylim((gain-1,+1))
plt.legend()
plt.show()
```

output

### PR DESCRIPTION
**Pull request by tjueterb**
### Which issue(s) are closed by this pull request?

Closes #392

### Changes proposed in this pull request:

Revised docstring to clarify that the characteristic frequency of the shelve filter in modes ``I`` and ``II`` can only be set if the absolute value of the gain is sufficiently large. Documentation build locally with sphinx looks ok to me.
### Labels

**documentation**, **dsp**

### CODE REVIEW

File: pyfar/dsp/filter/audiofilter.py [hunk 1]
```diff
@@ -114,16 +114,22 @@ def high_shelve(signal, frequency, gain, order, shelve_type='I',
...
-            defines the characteristic frequency at 3 dB if the gain is
-            positive and at -3 dB if the gain is negative.
+            Defines the characteristic frequency at 3 dB if the `gain` is
+            positive and at -3 dB if the `gain` is negative.
         ``'III'``
-            defines the characteristic frequency at gain/2 dB.
+            Defines the characteristic frequency at `gain`/2 dB.
+
+        For types ``I`` and ``II`` the absolute value of the `gain` must be
+        sufficiently large (>12 dB) to set the characteristic frequency
```
Comment (f-brinkmann) [inline pyfar/dsp/filter/audiofilter.py:130]: Should we be less explicit here? We have not tested it for gains between 6 and 12 dB. We could also write 
```suggestion
        sufficiently large (:math:`\gg3` dB) to set the characteristic frequency
```

File: pyfar/dsp/filter/audiofilter.py [hunk 2]
```diff
@@ -166,16 +172,22 @@ def low_shelve(signal, frequency, gain, order, shelve_type='I',
...
```
Comment (f-brinkmann) [inline pyfar/dsp/filter/audiofilter.py:188]: see above

**Review by f-brinkmann — COMMENTED** [review]
Thanks for the effort. I only had one minor question for discussion.

File: pyfar/dsp/filter/audiofilter.py (hunk 1 above)
Comment (tjueterb) [reply pyfar/dsp/filter/audiofilter.py:130]: I tested the error at the characteristic frequency for all gains, it's below 0.5 dB for gains $>\pm 9$ dB. The error does not depend on the characteristic frequency, nor the sampling rate.

shelve_error
```
import pyfar as pf
import numpy as np
import matplotlib.pyplot as plt

fs = 44100
f_c = 1000 # gain at characteristic frequency
order = 2
gains = np.linspace(24, -24, 100)
shelve_types = ['I', 'II']

impulse = pf.signals.impulse(n_samples=fs, sampling_rate=fs)
gain_f_c = np.zeros((len(shelve_types), len(gains))) # gain at f_c
expected_gain_f_c = np.zeros((len(shelve_types), len(gains))) # expected gain
gain_diff = np.zeros((len(shelve_types), len(gains))) # difference from expected gain

expected_gain_f_c[0,:] = gains - np.sign(gains) * 3
expected_gain_f_c[1,:] = np.sign(gains) * 3
# expected_gain_f_c[2,:] = gains / 2

fig, ax = plt.subplots(1,1)
for i, shelve_type in enumerate(shelve_types):
    for j, gain in enumerate(gains):
        ir = pf.dsp.filter.high_shelve(signal=impulse, frequency=f_c, gain=gain, order=order,
                                    shelve_type=shelve_type)
        idx = ir.find_nearest_frequency(f_c)
        gain_f_c[i, j] = 20*np.log10(np.abs(ir.freq[0,idx]))

    gain_diff[i, :] = np.abs(gain_f_c[i, :] - expected_gain_f_c[i, :])
    plt.plot(gains, gain_diff[i,:], label=f'shelve type {shelve_type}', alpha=0.8)

plt.hlines(0.5, np.min(gains), np.max(gains), colors='black', linestyles='dotted')
plt.vlines(-9, 0, 0.5, colors='black', alpha=0.3)
plt.vlines(+9, 0, 0.5, colors='black', alpha=0.3)
plt.xticks((-20, -9, 0, 9, 20))
plt.xlim((np.min(gains), np.max(gains)))
plt.ylim((0, 3))
plt.xlabel('gain in dB')
plt.ylabel('error at characteristic frequency in dB')
plt.legend()
plt.show()
```

**Review by f-brinkmann — APPROVED** [review]
Thanks for the effort!

**Review by ahms5 — APPROVED** [review]
Looks fine, thanks for the effort!

**Review by sikersten — APPROVED** [review]

**Pull request merged**